docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
+
+    let runAsync waitInSeconds =
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
+
+    // Output:
+    // Started Task
+    // Created Async
+    // Completed Task
+    // Completed Async
+
+### Async and Task Interop
+
+As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
```

Comment:
this needs a deeper explanation. Not sure how to do that succinctly though.
- It might be worth mentioning that F# async predates C# `await`
- everything ultimately runs on threads, managed in the .NET threadpool, and the low level APIs are all Task.
- C# majors on directly binding to Task stuff
- `task {` mirrors this
- `async {` is a higher level concept that offers a) flowing cancellation b) composition c) interop with `Task` d) a set of concurrency management primitives such as Async.Parallel etc
- Task has an analogous set of helpers such as Task.WhenAll

What you are saying is likely to sow confusion as StartAsTask vs RunSynchronously vs StartImmediateAsTask all end up running a Task, and continuations run on the threadpool etc. But there are details about how synchroniation contexts flow, cancellation flows, whether the initial work before the first wait point is part of the currrent piece fo work or only starts after a threadpool hop etc.

So while this statement is not strictly entirely wrong, you'll also find people willing to call it not even wrong...